server/src/main/java/org/opensearch/rest/action/cat/RestCatSegmentReplicationAction.java

```diff
@@ -0,0 +1,189 @@
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+public class RestCatSegmentReplicationAction extends AbstractCatAction {
+    @Override
+    public List<RestHandler.Route> routes() {
+        return unmodifiableList(
+            asList(new RestHandler.Route(GET, "/_cat/segment_replication"), new RestHandler.Route(GET, "/_cat/segment_replication/{index}"))
+        );
+    }
+
+    @Override
+    public String getName() {
+        return "cat_segment_replication_action";
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_cat/segment_replication\n");
+        sb.append("/_cat/segment_replication/{index}\n");
+    }
+
+    @Override
+    public BaseRestHandler.RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
+        final SegmentReplicationRequest segmentReplicationRequest = new SegmentReplicationRequest(
+            Strings.splitStringByCommaToArray(request.param("index"))
+        );
+        segmentReplicationRequest.timeout(request.param("timeout"));
+        segmentReplicationRequest.detailed(request.paramAsBoolean("detailed", false));
+        segmentReplicationRequest.activeOnly(request.paramAsBoolean("active_only", false));
+        segmentReplicationRequest.indicesOptions(IndicesOptions.fromRequest(request, segmentReplicationRequest.indicesOptions()));
+
+        return channel -> client.admin()
+            .indices()
+            .segment_replication(segmentReplicationRequest, new RestResponseListener<SegmentReplicationResponse>(channel) {
+                @Override
+                public RestResponse buildResponse(final SegmentReplicationResponse response) throws Exception {
+                    return RestTable.buildResponse(buildSegmentReplicationTable(request, response), channel);
+                }
+            });
+    }
+
+    @Override
+    protected Table getTableWithHeader(RestRequest request) {
+
+        Table t = new Table();
+        t.startHeaders()
+            .addCell("index", "alias:i,idx;desc:index name")
+            .addCell("shardId", "desc: shard Id")
+            .addCell("replication_id", "desc: replication Id")
+            .addCell("start_time", "default:false;alias:start;desc:segment replication start time")
+            .addCell("start_time_millis", "default:false;alias:start_millis;desc:segment replication start time in epoch milliseconds")
+            .addCell("stop_time", "default:false;alias:stop;desc:segment replication stop time")
+            .addCell("stop_time_millis", "default:false;alias:stop_millis;desc:segment replication stop time in epoch milliseconds")
+            .addCell("time", "alias:t,ti;desc:segment replication time")
+            .addCell("stage", "alias:st;desc:segment replication stage")
```

Comment:
does stage refer to the stage the event is currently in? For benchmarking or other tasks, could you clarify why we would need this information.

Since we already have fields for time taken in each particular stage is this redundant? The logic being that if the time taken for a particular stage is null we haven't reached that stage yet.

Reply:
Yes stage refers to current stage of the segment replication event. This will be useful on a API call made when segment replication is taking long or currently in-progress, then user can know exactly which stage is currently in-progress. 

Yes this will be useful for benchmarking. when calculating benchmarks we want to only include metrics of segrep events which are already completed. So we can say we aggregate metrics of a segrep event only when the segrep event's "stage"=="DONE", i.e only when segrep event is completed

I plan to put timing metric of each segment replication stage behind (detailed=true), so by default when (detailed=false) the API response will not have timing metric of each segrep event.